lib/logstash/inputs/gelf.rb

```diff
@@ -142,6 +142,27 @@ def self.coerce_timestamp(timestamp)
     timestamp.is_a?(BigDecimal) ? LogStash::Timestamp.at(timestamp.to_i, timestamp.frac * 1000000) : LogStash::Timestamp.at(timestamp)
   end
 
+  # from_json_parse uses the Event#from_json method to deserialize and directly produce events
+  def self.from_json_parse(json)
+    LogStash::Event.from_json(json).each { |event| event }
+  rescue LogStash::Json::ParserError
+    LogStash::Event.new("message" => json, "tags" => ["_jsonparsefailure"])
```

Comment:
I suggest you defines constants for the `"message"`, `"tags"` and `"_jsonparsefailure"` strings. 
Ideally we should have them defines higher up in logstash but they don't so the next best this for now is to define them as constants here.